<a href="https://colab.research.google.com/github/Hed-NK/Concordia-Bootcamps-Data-Science-Admissions-Challenge/blob/main/DataScienceAdmissionChallenge.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

#Step 1: Getting Started

In [1]:
import pandas as pd

#Step 2: Loading and Exporing the Data

In [2]:
from google.colab import files
uploaded = files.upload()
df = pd.read_csv('results.csv')
df.info()

Saving results.csv to results.csv
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 45315 entries, 0 to 45314
Data columns (total 9 columns):
 #   Column      Non-Null Count  Dtype 
---  ------      --------------  ----- 
 0   date        45315 non-null  object
 1   home_team   45315 non-null  object
 2   away_team   45315 non-null  object
 3   home_score  45315 non-null  int64 
 4   away_score  45315 non-null  int64 
 5   tournament  45315 non-null  object
 6   city        45315 non-null  object
 7   country     45315 non-null  object
 8   neutral     45315 non-null  bool  
dtypes: bool(1), int64(2), object(6)
memory usage: 2.8+ MB


#Step 3: Data Analysis

#Q1: What is the average number of goals for the home team?

In [3]:
print(df['home_score'].mean())

1.7393136930376256


The answer is 1.7393136930376256

#Q2: What is the average number of goals for the away team?

In [4]:
print(df['away_score'].mean())

1.1782412004854905


The answer is 1.1782412004854905

#Q3: What is the country with the highest overall home score?

In [11]:
grouped_data_by_Country = df.groupby('country')['home_score'].sum()
sum_of_home_scores_for_each_country = grouped_data_by_Country.idxmax()
print("Country with the highest sum of home scores:", sum_of_home_scores_for_each_country)

Country with the highest sum of home scores: United States


The answer is Country with the highest sum of home scores: United States

#Q4: What are the top 3 type of tournaments?

In [12]:
tournament_count = df['tournament'].value_counts()
top_three_tournaments = tournament_count.head(3)
print(top_three_tournaments)

Friendly                        17773
FIFA World Cup qualification     8016
UEFA Euro qualification          2815
Name: tournament, dtype: int64


The answer is
Friendly                        17773
FIFA World Cup qualification     8016
UEFA Euro qualification          2815
Name: tournament, dtype: int64

#Q5: Which country has the highest overall FIFA World Cup goals?

In [13]:
fifa_df = df[df['tournament'] == 'FIFA World Cup']
fifa_df['total_score'] = fifa_df['home_score'] + fifa_df['away_score']
grouped_data_by_Country = fifa_df.groupby('country')['total_score'].sum()
country_with_highest_total_score = grouped_data_by_Country.idxmax()
print("Country with the highest overall FIFA World Cup goals:", country_with_highest_total_score)

Country with the highest overall FIFA World Cup goals: Brazil


<ipython-input-13-c9f9413d3a2d>:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  fifa_df['total_score'] = fifa_df['home_score'] + fifa_df['away_score']


The answer is Country with the highest overall FIFA World Cup goals: Brazil